Add `gcompris` educational package

guibou opened the pull request:
###### Motivation for this change

Add `gcompris`, which is a mandatory educational package for kids.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X ] Tested execution of all binary files (usually in `./result/bin/`)
- [ X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

guibou committed:
qmlbox2d: init at 2018-03-16

This is a dependency for `gcompris`

guibou committed:
gcompris: init at 0.90

- uses qt 5.9 (qt 5.10 gives runtime error)
- add a desktop file

jtojnar commented on pkgs/games/gcompris/default.nix:
```diff
@@ -0,0 +1,33 @@
+
+  cmakeFlags = "-DQML_BOX2D_LIBRARY=${qmlbox2d}/${qtbase.qtQmlPrefix}/Box2D.2.0";
+
+  nativeBuildInputs = [ cmake makeWrapper ];
```
Please add ninja when using cmake.

guibou replied:
I was not aware of the runtime detection of `ninja` by `cmake`. Thank you. Fixed.

jtojnar commented on pkgs/games/gcompris/default.nix:
```diff
@@ -0,0 +1,33 @@
+    sha256 = "1i5adxnhig849qxwi3c4v7r84q6agx1zxkd69fh4y7lcmq2qiaza";
+  };
+
+  enableParallelBuilding = true;
```
cmake builds in parallel by default.

guibou replied:
Explicit is better than implicit no? Do you really want me to remove it?

nlewo replied:
Hmm, there are a lot of other attributes that are implicit:) I also think we can let default values to be applied.

guibou replied:
Done.

nlewo commented:
I've got this error:
```
 /nix/store/c90qjb1n7p7gkj4xf29rj8lx84jbm92h-gcompris-0.90/bin/gcompris-qt 
QCursor: Cannot create bitmap cursor; invalid bitmap(s)
exeCount set to:  1
QQmlApplicationEngine failed to load component
qrc:/gcompris/src/core/main.qml:22 module "QtQuick.Controls" is not installed
qrc:/gcompris/src/core/main.qml:23 module "QtQuick.Window" is not installed
qrc:/gcompris/src/core/main.qml:21 module "QtQuick" is not installed
qrc:/gcompris/src/core/main.qml:22 module "QtQuick.Controls" is not installed
qrc:/gcompris/src/core/main.qml:23 module "QtQuick.Window" is not installed
qrc:/gcompris/src/core/main.qml:21 module "QtQuick" is not installed
qrc:/gcompris/src/core/main.qml:22 module "QtQuick.Controls" is not installed
qrc:/gcompris/src/core/main.qml:23 module "QtQuick.Window" is not installed
qrc:/gcompris/src/core/main.qml:21 module "QtQuick" is not installed

Cannot mix incompatible Qt library (version 0x50901) with this library (version 0x50903)
Aborted
```

Any idea about what is happening?

guibou commented:
@nlewo try to install the package in your env : `nix-env -i /nix/store/c90qjb1n7p7gkj4xf29rj8lx84jbm92h-gcompris-0.90/`. The nixpkgs manual told so, I don't know why...

nlewo commented:
@guibou ok, it work with `nix-env`.

guibou commented:
I fixed the `enableParallelBuilding` as requested by @jtojnar  and added a `.desktop` file entry. However this PR does not see the changes, I wonder why. I will let a few time to github to see if it refreshs itself and if not, I'll open a new PR.

nlewo commented:
@guibou looks like the PR has now been updated.
@GrahamcOfBorg build gcompris

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.37377&attempt_id=541854d7-2d9a-4d03-9212-007f3f55fea6)

Attempted: gcompris

<details><summary>Partial log (click to expand)</summary><p>

```
   Compiling backtrace v0.3.4
   Compiling rustc_const_math v0.0.0 (file:///build/rustc-1.24.0-src/src/librustc_const_math)
   Compiling proc_macro v0.0.0 (file:///build/rustc-1.24.0-src/src/libproc_macro)
   Compiling syntax_ext v0.0.0 (file:///build/rustc-1.24.0-src/src/libsyntax_ext)
building of '/nix/store/whsyh46wr0zfkw420xcc8ffgahszsd80-rustc-1.24.0.drv' timed out after 3600 seconds
cannot build derivation '/nix/store/nfjkyviyyy9ly6hq1pj5wzy6qxmd491n-cargo-0.24.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/bbqaq71xj30lggbbbf6calm3g7g56c02-librsvg-2.42.2.drv': 2 dependencies couldn't be built
cannot build derivation '/nix/store/z97wv2682mjz6bvgffwa3994ryav1vgs-gst-plugins-bad-1.12.3.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/npjyjxy73wzazv90r7gkhwj643v4g67b-gcompris-0.90.drv': 1 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/npjyjxy73wzazv90r7gkhwj643v4g67b-gcompris-0.90.drv' failed
```
</p></details>

nlewo commented:
@guibou could you either fix the `aarch64` build or restrict the platforms to `x86` only?

guibou commented:
@nlewo 

Done, set to `[ "i686-linux" "x86_64-linux" ]`.

nlewo commented:
@GrahamcOfBorg build qmlbox2d

GrahamcOfBorg commented:
No attempt on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.37377&attempt_id=a7177cc0-2182-42bd-a445-9143e5c14fea)

The following builds were skipped because they don't evaluate on x86_64-linux: qmlbox2d

<details><summary>Partial log (click to expand)</summary><p>

```
Cannot nix-instantiate `qmlbox2d' because:
[31;1merror:[0m attribute 'qmlbox2d' in selection path 'qmlbox2d' not found

```
</p></details>

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.37377&attempt_id=2ce7c9ad-ac59-406c-b430-c128e1674c4e)

The following builds were skipped because they don't evaluate on aarch64-linux: qmlbox2d

<details><summary>Partial log (click to expand)</summary><p>

```
Cannot nix-instantiate `qmlbox2d' because:
[31;1merror:[0m attribute 'qmlbox2d' in selection path 'qmlbox2d' not found

```
</p></details>

nlewo commented:
Thanks... from my child:)
